crates/wash-lib/src/start/github.rs

```diff
@@ -87,3 +90,41 @@ where
         .await
         .map_or(false, |m| m.is_file())
 }
+
+/// Helper function to set up a reqwest client for performing the download
+fn get_download_client() -> reqwest::Client {
+    let proxy_username = std::env::var("WASH_PROXY_USERNAME").unwrap_or_default();
+    let proxy_password = std::env::var("WASH_PROXY_PASSWORD").unwrap_or_default();
+    // TODO(joonas): should we consider system proxy configuration here, or just explicit config?
+    let http_proxy_addr = match std::env::var("HTTP_PROXY") {
+        Ok(addr) => addr,
+        Err(_) => std::env::var("http_proxy").unwrap_or_default(),
+    };
+    let https_proxy_addr = match std::env::var("HTTPS_PROXY") {
+        Ok(addr) => addr,
+        Err(_) => std::env::var("https_proxy").unwrap_or_default(),
+    };
+
+    let mut builder = reqwest::ClientBuilder::default()
+        .user_agent(DOWNLOAD_CLIENT_USER_AGENT)
+        .with_native_certificates();
+
+    if !http_proxy_addr.is_empty() {
+        let mut proxy = reqwest::Proxy::http(http_proxy_addr).expect("failed to create http proxy");
+        if !proxy_username.is_empty() && !proxy_password.is_empty() {
+            proxy = proxy.basic_auth(&proxy_username, &proxy_password);
+        }
+        builder = builder.proxy(proxy);
+    }
+
+    if !https_proxy_addr.is_empty() {
+        let mut proxy =
+            reqwest::Proxy::https(https_proxy_addr).expect("failed to create https proxy");
+        if !proxy_username.is_empty() && !proxy_password.is_empty() {
+            proxy = proxy.basic_auth(&proxy_username, &proxy_password);
+        }
+        builder = builder.proxy(proxy);
+    }
+
+    builder.build().expect("failed to build HTTP client")
```

Comment:
would be nice to bubble up the error here and not turn it into a panic!

Reply:
that's fair, and I agree that would probably be the right thing to do here, my intent was to preserve the [current/existing implementation's handling](https://github.com/wasmCloud/wasmCloud/blob/main/crates/core/src/tls.rs#L65-L70), but I agree that bubbling up the error would be better.

Reply:
I've made these into `Result<...>` instead, thanks for the callout!